Homework2/Homework2/Program.cs

```diff
@@ -0,0 +1,183 @@
+        {
+            var numbers = string.Join(" ",array);
+            Console.WriteLine(numbers);
+        }
+
+        static void SortArray(int[] array) // Sort an array
+        {
+            int[] NewArray = new int[array.Length];
+            Array.Copy(array, NewArray, array.Length);
+
+            for(int i = NewArray.Length - 1; i > 0; i--)
+            {
+                for(int j = 0; j < i; j++)
+                {
+                    if(NewArray[j] > NewArray[j+1])
+                    {
+                        int temp = NewArray[j + 1];
+                        NewArray[j + 1] = NewArray[j];
+                        NewArray[j] = temp;
+                    }
+                }
+            }
+
+            Console.WriteLine("Before: ");
+            PrintArray(array);
+            Console.WriteLine("After: ");
+            PrintArray(NewArray);
+        }
+
+        #region AddToArray
+
+        static void AddToStart(int num, int[] array) // Add element at the start of an array
+        {
+            AddToArrayPosition(num, 0, array);
+        }
+
+        static void AddToEnd(int num, int[] array) // Add element at the end of an array
+        {
+            AddToArrayPosition(num, array.Length, array);
+        }
+
+        static void AddToArrayPosition(int num, int position, int[] array) // Add element at any position of an array
+        {
+            int[] NewArray = new int[array.Length + 1];
+            int temp = 0;
+
+            for (int index = 0; index < NewArray.Length; index++)
+            {
+                if (index != position)
+                {
+                    NewArray[index] = array[temp++];
+                }
+                else
+                {
+                    NewArray[index] = num;
+                }
+            }
+
+            Console.WriteLine("\nBefore: ");
+            PrintArray(array);
+            Console.WriteLine("\nAfter: ");
+            PrintArray(NewArray);
+        }
+
+        #endregion
+
+        #region RemoveFromArray
+
+        static void RemoveAtStart(int[] array) // Remove element at the start of an array
+        {
+            RemoveAtAny(array, 0);
+        }
+
+        static void RemoveAtEnd(int[] array) // Remove element at the end of an array
+        {
+            RemoveAtAny(array, array.Length);
+        }
+
+        static void RemoveAtAny(int[] array, int position) // Remove element at a given index of an array
+        {
+            int[] NewArray = new int[array.Length - 1];
+            int temp = 0;
+            for (int index = 0; index < array.Length; index++)
+            {
+                if (index != position && temp != NewArray.Length)
+                {
+                    NewArray[temp++] = array[index];
+                }
+            }
+
+            Console.WriteLine("\nBefore: ");
+            PrintArray(array);
+            Console.WriteLine("\nAfter: ");
+            PrintArray(NewArray);

```

Comment:
This console writeline is a side effect. Unless debugging, don't write lines like this

Reply:
If you do want to, write it outside of a function